### DISCUSSION TITLE
convert longitude to date

### DISCUSSION THREAD

**Discussion by arumdyaharum** [cosinekitty/astronomy#306] [Q&A] [answered]
is it possible to convert from longitude to date?

for context, human design uses two astrology charts. the first chart is from our birthtime, the second chart is 88° backward from birthtime sun longitude. I get sun correctly but I need sun date in order to know the rest of the second chart.

**Comment by cosinekitty** [comment]
Sorry, I don't understand this question. I keep re-reading this but I can't figure out what you mean.

**Reply by arumdyaharum** [reply]
So I'm trying to create human design chart that looks like below. It uses 2 astronomical position, design side and personality side. personality side is created from the time when we born. design side is created from 88° backward or around 85-95 days before we born.

my approach is I get the longitude of whole planetary position of the personality side from birthtime first. and then I get design sun by subtract personality sun by 88. in order to know the rest planetary position of the design side, I need to know the date of the design side.

I asked people from human design community and they said I can search manually around 85-95 days before birthtime and look at the closest longitude with accuracy of 5 decimal places.

here's my code.

```js
function geoVectorToDegree (date, planet) {
  const vector = Astronomy.GeoVector(PLANETS[planet], Astronomy.MakeTime(date), true);
  let longitude = Astronomy.Ecliptic(vector).elon;
  // Convert 180 format to 360 format
  if (longitude < 0) longitude = longitude + 360;
  return longitude;
}

function hdCreation (date, isDesign = false) {
  const sunDegree = geoVectorToDegree(date, "sun");
  const sun = degreeToHdGates(isDesign ? sunDegree - 88 : sunDegree);
  const earth = degreeToHdGates(isDesign ? sunDegree - 88 + 180 : sunDegree + 180);
}
```

<img src="https://images.squarespace-cdn.com/content/v1/5c2e3e0050a54fad82f872df/1614646244661-F15UJ1XTTZ81ZDSTMVY7/Screen+Shot+2021-03-01+at+7.49.40+PM.png?format=1000w"/>

**Comment by cosinekitty** [comment, marked as answer]
OK, that explanation helps. I recommend calling [SunPosition](https://github.com/cosinekitty/astronomy/tree/master/source/js#SunPosition) to find the Sun's longitude at the birth date. It returns type [EclipticCoordinates](https://github.com/cosinekitty/astronomy/tree/master/source/js#eclipticcoordinates), which contains the Sun's ecliptic longitude in the variable `elon`.

Once you have that, subtract 88 degrees, and pass the result to [SearchSunLongitude](https://github.com/cosinekitty/astronomy/tree/master/source/js#searchsunlongitudetargetlon-datestart-limitdays--astrotime--null). The resulting code could look something like this:

```javascript
const birth = Astronomy.MakeTime(new Date('2023-06-12T00:00:00Z'));
const sun = Astronomy.SunPosition(birth);
const target = ((sun.elon - 88) + 360) % 360;
const design = Astronomy.SearchSunLongitude(target, birth.AddDays(-100), 100);
console.log(design.date);    //  2023-03-13T16:03:31.528Z
```

**Comment by arumdyaharum** [comment]
Thank you @cosinekitty. I have one more question regarding lunar nodes. astrology and human design treat lunar nodes as a planetary position so it's important to know the longitude of lunar nodes through time. am I need to create another discussion or create new issue?

**Reply by cosinekitty** [reply]
Yes, please start a new discussion. We keep the discussions around so other people can read them later, so it helps to keep the topics organized. Thank you!
